### PR TITLE
Implemented msodde scanning in DOC_INFO. Fixes #875

### LINKED ISSUE #875: add msodde tool to Doc_Info analyzer
See: https://github.com/decalage2/oletools/wiki/msodde

We should add this tool in the pipeline of tools to run to detect anomalies in Office files.

Follow this thread in [Twitter](https://twitter.com/decalage2/status/1488481984242462728) to get more details on real use cases (also for testing the solution)

### PR DESCRIPTION
**Pull request by devmrfitz**
# Description

Implemented msodde scanning in DOC_INFO
Screenshot of test result:
image

## Related issues
closes #875 

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue).
- [X] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).

# Checklist

- [X] I have read and understood the rules about [how to Contribute](https://intelowl.readthedocs.io/en/latest/Contribute.html) to this project
- [X] The pull request is for the branch `develop`
- [ ] A new analyzer or connector was added, in which case:
    - [ ] [Usage](https://github.com/intelowlproject/IntelOwl/blob/master/docs/source/Usage.md) file was updated.
    - [ ] [Advanced-Usage](./Advanced-Usage.md) was updated (in case the analyzer/connector provides additional optional configuration).
    - [ ] Secrets were added in [env_file_app_template](https://github.com/intelowlproject/IntelOwl/blob/master/docker/env_file_app_template), [env_file_app_ci](https://github.com/certego/IntelOwl/blob/master/docker/env_file_app_ci) and in the [Installation](./Installation.md) docs, if necessary.
    - [ ] If the analyzer/connector requires mocked testing, `_monkeypatch()` was used in it's class to apply the necessary decorators.
    - [ ] If a File analyzer was added, it's name was explicitly defined in [test_file_scripts.py](https://github.com/intelowlproject/IntelOwl/blob/master/tests/analyzers_manager/test_file_scripts.py) (not required for Observable Analyzers).
- [ ] If external libraries/packages with restrictive licenses were used, they were added in the [Legal Notice](https://github.com/certego/IntelOwl/blob/master/.github/legal_notice.md) section.
- [X] The tests gave 0 errors.
- [X] Linters (`Black`, `Flake`, `Isort`) gave 0 errors. If you have correctly installed [pre-commit](https://intelowl.readthedocs.io/en/latest/Contribute.html#how-to-start-setup-project-and-development-instance), it does these checks and adjustments on your behalf.
- [X] The commits were squashed into a single one (optional, they will be squashed anyway by the maintainer)
  
### Important Rules
- If your changes decrease the overall tests coverage (you will know after the Codecov CI job is done), you should add the required tests to fix the problem
- Everytime you make changes to the PR and you think the work is done, you should explicitly ask for a review

# Real World Example

Please delete if the PR is for bug fixing.
Otherwise, please provide the resulting raw JSON of a finished analysis (and, if you like, a screenshot of the results). This is to allow the maintainers to understand how the analyzer works.

### CODE REVIEW

**Comment by devmrfitz** [comment]
@mlodic Is this the correct way to do the task?

File: api_app/analyzers_manager/file_analyzers/doc_info.py [hunk 1]
```diff
@@ -134,6 +135,17 @@ def run(self):
...
         results["olevba"] = self.olevba_results
 
+        try:
+            msodde_result = msodde_process_maybe_encrypted(self.filepath)
+        except Exception as e:
+            error_message = f"job_id {self.job_id} msodde parser failed. Error: {e}"
+            logger.exception(error_message)
+            self.report.errors.append(error_message)
+            self.report.save()
+            msodde_result = "Error: {}".format(e)
```
Comment (mlodic) [inline api_app/analyzers_manager/file_analyzers/doc_info.py:145]: please use the more recent f-format strings

File: api_app/analyzers_manager/file_analyzers/doc_info.py [hunk 2]
```diff
@@ -134,6 +135,17 @@ def run(self):
 
         results["olevba"] = self.olevba_results
 
+        try:
+            msodde_result = msodde_process_maybe_encrypted(self.filepath)
```
Comment (mlodic) [inline api_app/analyzers_manager/file_analyzers/doc_info.py:139]: you could also pass the argument `password` with `self.passwords_to_check`

**Review by mlodic — CHANGES_REQUESTED** [review]
thanks for the PR! the code makes sense. A question: can you share the md5 of the sample you tried?

**Comment by devmrfitz** [comment]
> thanks for the PR! the code makes sense. A question: can you share the md5 of the sample you tried?

[Here](https://file.io/fHVyTTod5H6V) is a link to the file I generated. It's md5sum is `12e274121124558a2d2356d9dbdec1a5`

**Review by mlodic — APPROVED** [review]

**Comment by mlodic** [comment]
that's great!

**Pull request merged**